connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/admin/LoggingResourceTest.java

```diff
@@ -0,0 +1,189 @@
+        b.setLevel(Level.INFO);
+        when(loggingResource.currentLoggers()).thenReturn(loggers(a, b));
+        when(loggingResource.rootLogger()).thenReturn(root);
+        when(loggingResource.getLogger(any())).thenCallRealMethod();
+        loggingResource.getLogger("c");
+    }
+
+    @Test
+    public void setLevelTest() {
+        LoggingResource loggingResource = mock(LoggingResource.class);
+        Logger root = new Logger("root") {
+        };
+        root.setLevel(Level.ERROR);
+        Hierarchy hierarchy = new Hierarchy(root);
+        Logger p = hierarchy.getLogger("a.b.c.p");
+        Logger x = hierarchy.getLogger("a.b.c.p.X");
+        Logger y = hierarchy.getLogger("a.b.c.p.Y");
+        Logger z = hierarchy.getLogger("a.b.c.p.Z");
+        Logger w = hierarchy.getLogger("a.b.c.s.W");
+        x.setLevel(Level.INFO);
+        y.setLevel(Level.INFO);
+        z.setLevel(Level.INFO);
+        w.setLevel(Level.INFO);
+        when(loggingResource.currentLoggers()).thenReturn(loggers(x, y, z, w));
+        when(loggingResource.lookupLogger("a.b.c.p")).thenReturn(p);
+        when(loggingResource.rootLogger()).thenReturn(root);
+        when(loggingResource.setLevel(any(), any())).thenCallRealMethod();
+        List<String> modified = (List<String>) loggingResource.setLevel("a.b.c.p", Collections.singletonMap("level", "DEBUG")).getEntity();
+        assertEquals(4, modified.size());
+        assertEquals(Arrays.asList("a.b.c.p", "a.b.c.p.X", "a.b.c.p.Y", "a.b.c.p.Z"), modified);
+        assertEquals(p.getLevel(), Level.DEBUG);
+        assertEquals(x.getLevel(), Level.DEBUG);
+        assertEquals(y.getLevel(), Level.DEBUG);
+        assertEquals(z.getLevel(), Level.DEBUG);
+    }
+
+    @Test
+    public void setRootLevelTest() {
+        LoggingResource loggingResource = mock(LoggingResource.class);
+        Logger root = new Logger("root") {
+        };
+        root.setLevel(Level.ERROR);
+        Hierarchy hierarchy = new Hierarchy(root);
+        Logger p = hierarchy.getLogger("a.b.c.p");
+        Logger x = hierarchy.getLogger("a.b.c.p.X");
+        Logger y = hierarchy.getLogger("a.b.c.p.Y");
+        Logger z = hierarchy.getLogger("a.b.c.p.Z");
+        Logger w = hierarchy.getLogger("a.b.c.s.W");
+        x.setLevel(Level.INFO);
+        y.setLevel(Level.INFO);
+        z.setLevel(Level.INFO);
+        w.setLevel(Level.INFO);
+        when(loggingResource.currentLoggers()).thenReturn(loggers(x, y, z, w));
+        when(loggingResource.lookupLogger("a.b.c.p")).thenReturn(p);
+        when(loggingResource.rootLogger()).thenReturn(root);
+        when(loggingResource.setLevel(any(), any())).thenCallRealMethod();
+        List<String> modified = (List<String>) loggingResource.setLevel("@root", Collections.singletonMap("level", "DEBUG")).getEntity();
+        assertEquals(5, modified.size());
+        assertEquals(Arrays.asList("a.b.c.p.X", "a.b.c.p.Y", "a.b.c.p.Z", "a.b.c.s.W", "root"), modified);
```

Comment:
Can we also check the actual logger's levels here, just like in `setLevelTest()`?